src/app/EventLogging.h

```diff
@@ -53,15 +55,55 @@ class EventLogger : public EventLoggingDelegate
  * context tags to be interpreted within the schema identified by
  * `ClusterID` and `EventId`.
  *
- * @param[in] apDelegate The EventLoggingDelegate to serialize the event data
+ * LogEvent has 3 variant, for aEventData that have method GetFabricIndex with return type FabricIndex or
+ * DataModel::Nullable<chip::FabricIndex>, it would use first two logEvent function twhich can process the fabric index, for
+ * aEventData that don't have GetFabricIndex, it would use tird LogEvent function.
```

Comment:
When would a null FabricIndex be returned?  How is that different from a FabricIndex of 0?

(Also, is there a reason this is using Nullable instead of Optional to mean "no fabric index"?  Nullable is meant to be used for the spec concept of nullable; that being things that get encoded as a TLV null.  Do we do that for event fabric indices?)

Reply:
from spec, https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/686ca55b4764cc6e1fccd348671a7360a5f6828e/src/app_clusters/DoorLock.adoc#534-fabricindex."The fabric index of the fabric that performed the lock operation. This SHALL be null if there is no fabric that can be determined for the given operation source. This SHALL NOT be null if the operation source is "Remote".,
For null, I think it means this event can be retrieved by read handler, and there is no fabric restriction on it. 0 is still meaningful fabric.

Reply:
Updated document.

Reply:
Per offline chat, we would check for a field that is of type fabric_idx and not nullable which could be used for fabric-scoped event, currently acl is fabric scope one. updated the code